### PR TITLE
Sort, bugs and developer guide

### PR DESCRIPTION
**Pull request by Donovan9617**
This PR consists of features that:
- Enable user to sort projects by deadline, issue counts, and name
- Enable user to sort issues by deadline, priority
- Error handling, verification, logic, and UI displays for sorting projects and issues
- Enable deleting projects by id instead of list position
- Enable optional parameters for `editIssueCommand`
- Edit DG for add command with content and UML diagram

Note:
- Sort keeps the order of entity the next time the app is opened

Approach for Sort:
- Create a view list for projects/issues which assumes initial list from memory. Sort projects/issues in the view list based on specified key and order. Attach this updated view list to the root of the main window.

### CODE REVIEW

**Review by crvstalphua — APPROVED** [review]
Had one comment but LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -196,12 +196,54 @@ The following activity diagram summarizes what happens when a user executes a li
...
+
+#### Design considerations:
+
+**Aspect: Add command access to the model: **
+
+**Alternative 1: ** Refactor `ProjectCommandParser:parseAddProjectCommand`, `IssueCommandParser:parseAddIssueCommand` and `ClientCommandParser:parseAddClientCommand` to have access to the Model.
+ * Pros: Mappings could be performed within the parser which fitted its responsibility.
+ * Cons: May result in extra coupling between Parser class and Model class.
+
+**Alternative 2: (current choice)** Only `AddProjectCommand:execute`, `IssueCommandParser:execute` and `ClientCommandParser:execute` have access to the Model.
```
Comment (ConradLew) [inline docs/DeveloperGuide.md:242]: Minor nit - current choice might be better off documented as the first alternative

File: src/main/java/seedu/address/model/project/Project.java [hunk 2]
```diff
@@ -123,6 +132,57 @@ public void removeClient() {
...
+     *
+     * 0 for alphabetical order and 1 for reverse alphabetical order
+     *
+     * @param num input param to validate
+     * @return true if input is a 0 or 1
+     */
+    public static boolean isValidNameSortKey(String num) {
+        try {
+            int number = Integer.parseInt(num);
+            return number == 0 | number == 1;
```
[the remark this answers is not in the document]
Comment (Donovan9617) [reply src/main/java/seedu/address/model/project/Project.java:180]: Changed to ||

File: src/main/java/seedu/address/logic/commands/issue/SortIssueCommand.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        String sortKeyString = "";
+
+        if (sortKey.equals(PREFIX_DEADLINE)) {
+            model.sortIssuesByDeadline(sortOrder);
+            sortKeyString = "deadline";
+        }
+
+        if (sortKey.equals(PREFIX_PRIORITY)) {
+            model.sortIssuesByPriority(sortOrder);
+            sortKeyString = "priority.";
```
[the remark this answers is not in the document]
Comment (Donovan9617) [reply src/main/java/seedu/address/logic/commands/issue/SortIssueCommand.java:60]: Will keep it consistent by adding . for all

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Donovan9617) [reply docs/DeveloperGuide.md:242]: Sure current choice to be the first option

File: src/main/java/seedu/address/logic/commands/project/DeleteProjectCommand.java [hunk 4]
```diff
@@ -42,31 +42,29 @@ public CommandResult execute(Model model, Ui ui) throws CommandException {
...
+            if (p.getProjectId().getIdInt() == targetIndex.getOneBased()) {
+                List<Issue> listOfIssuesToDelete = p.getIssueList();
+                for (Issue i : listOfIssuesToDelete) {
+                    model.deleteIssue(i);
+                }
+                Client projectClient = p.getClient();
+                if (!projectClient.isEmpty()) {
+                    projectClient.removeProject(p);
+                    if (projectClient.getProjectListSize() == 0) {
+                        model.getAddressBook().getClientList().remove(projectClient);
```
[the remark this answers is not in the document]
Comment (Donovan9617) [reply src/main/java/seedu/address/logic/commands/project/DeleteProjectCommand.java:55]: Ah good catch, I took reference from `deleteIssueCommand` so will change this and that one too

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -196,12 +196,54 @@ The following activity diagram summarizes what happens when a user executes a li
...
-_{Explain here how the data archiving feature will be implemented}_
+#### Add Project Command
+Compulsory prefixes: n/<valid name>
+Optional prefixes: cid/<valid client id>, r/<valid repository>, d/<valid deadline>
+Example Use: `project -a n/John cid/1 r/JohnDoe/tp d/2022-03-05`
 
+#### Add Issue Command
+Compulsory prefixes: n/<valid name>, desc/<valid description>
+Optional prefixes: pid/<valid project id> d/<valid deadline> p/<valid priority>
+Example Use: `issue pid/1 desc/To create a person class which stores all relevant person data d/2022-12-10 p/0`
```
Comment (crvstalphua) [inline docs/DeveloperGuide.md:217]: Issue doesn't use name, and project id is compulsory 
```suggestion
Compulsory prefixes: pid/<valid project id>, desc/<valid description>
Optional prefixes: d/<valid deadline> p/<valid priority>
Example Use: `issue pid/1 desc/To create a person class which stores all relevant person data d/2022-12-10 p/0`
```

**Review by crvstalphua — APPROVED** [review]
just some comments to be edited then lgtm

**Comment by Donovan9617** [comment]
#103

**Pull request merged**